portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,47 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// the indexes recording the picture 
+// the slides in gallery.html is showing
+var slide1 = 0;
```

Comment:
Here and elsewhere, prefer to use const and let over var: https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations